metals/src/main/scala/scala/meta/internal/metals/OnTypeFormattingProvider.scala

```diff
@@ -0,0 +1,73 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.{DocumentOnTypeFormattingParams, Range, TextEdit}
+
+import scala.meta.internal.mtags.Semanticdbs
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.inputs.Input
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token.Constant
+
+final class OnTypeFormattingProvider(
+    semanticdbs: Semanticdbs,
+    buffer: Buffers
+)(implicit ec: ExecutionContext) {
+
+  val tripleQuote = """\u0022\u0022\u0022"""
+  val space = " "
+
+  private def indent(toInput: String, pos: meta.Position): String = {
+    val beforePos = toInput.substring(0, pos.start)
+    val lastPipe = beforePos.lastIndexOf("|")
+    val lastNewline = beforePos.lastIndexOf("\n", lastPipe)
+    val indent = beforePos.substring(beforePos.lastIndexOf("\n")).length
+    val length = toInput.substring(lastNewline, lastPipe).length
+    space * (length - indent)
+  }
+
+  private def isLongString(text: String, token: Token): Boolean = {
+    text.substring(token.start, token.start + 3).equals(tripleQuote)
+  }
+
+  private def inToken(pos: meta.Position, token: Token): Boolean = {
+    pos.start >= token.start && pos.end <= token.end
+  }
+
+  private def pipeInScope(pos: meta.Position, text: String): Boolean = {
+    val firstNewline = text.substring(0, pos.start).lastIndexOf("\n")
+    val lastNewline =
+      text.substring(0, firstNewline).lastIndexOf("\n")
+    text
+      .substring(lastNewline + 1, pos.start)
+      .contains("|")
+  }
+
+  def format(
+      params: DocumentOnTypeFormattingParams
+  ): Future[Option[TextEdit]] = {
+    val source = params.getTextDocument.getUri.toAbsolutePath
+    var edit: Option[TextEdit] = None
+    if (source.exists) {
+      val range = new Range(params.getPosition, params.getPosition)
+      val sourceText = buffer.get(source).getOrElse("")
+      val pos = params.getPosition.toMeta(
+        Input.VirtualFile(source.toString(), sourceText)
+      )
+      if (pipeInScope(pos, sourceText)) {
+        val tokens =
+          Input.VirtualFile(source.toString(), sourceText).tokenize.toOption
+        tokens.flatMap { tokens =>
+          tokens.collectFirst {
+            case token: Constant.String
+                if inToken(pos, token) && isLongString(sourceText, token) =>
+              edit = Some(new TextEdit(range, indent(sourceText, pos) + "|"))
```

Comment:
Let's just return the `new TextEdit(...)` here. as a result, the `tokens.flatMap` on line 62 will be returning this edit.
```scala
if (pipeInScope(pos, sourceText)) {
  val tokens = Input.VirtualFile(...)
  tokens.flatMap { tokens => tokens.collectFirst{ case (...) if (...) => new TextEdit(...) } }
} else Future.successfull(None)
```
or we could do:
```scala
tokens match {
  case None => 
    Future.successful(None)
  case tokens => 
    tokens.collectFirst{ case (...) if (...) => new TextEdit(...) }
}
```

whatever you find more readable

Reply:
? how can we return Future.succesful(None) when we want to return Future[TextEdit]? wouldn't a None be more fitting for Option[TextEdit]?